packages/babel-parser/src/parser/expression.js

```diff
@@ -1605,14 +1605,19 @@ export default class ExpressionParser extends LValParser {
     node.quasis = [curElt];
     while (!curElt.tail) {
       this.expect(tt.dollarBraceL);
-      node.expressions.push(this.parseExpression());
+      node.expressions.push(this.parseTemplateInterpolation());
```

Comment:
Can we extend `parseTemplate` in `typescript` plugin? since it seems to me that `TemplateInterpolation` is a TS-only concept.

Reply:
Isn't it the same concept as for js? i.e. <code>`a${x}b`</code> is valid both in TS and in JS.
Maybe if I call it `parseTemplateSubstitution`, which is the language [used in the spec](https://tc39.es/ecma262/#prod-SubstitutionTemplate)?